gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/pages/GradebookPage.java

```diff
@@ -399,65 +408,15 @@ public String getCssClass() {
 		modelData.put("assignments", assignments);
 		modelData.put("categories", categories);
 
-		this.form.add(new GradebookSpreadsheetFixedTables("fixedHeader", Model.ofMap(modelData)));
-
-		if (settings.isCategoriesEnabled()) {
-			// Pre-sort columns so the JavaScript doesn't have to
-			// by putting the category total columns after assignments
-			// in that category (which are already grouped by category).
-			// If categories are not enabled, these average columns
-			// will display as the last columns in the table.
-			Collections.sort(cols, new Comparator<IColumn>() {
-				@Override
-				public int compare(IColumn col1, IColumn col2) {
-					IModel model1 = ((AbstractColumn) col1).getDisplayModel();
-					IModel model2 = ((AbstractColumn) col2).getDisplayModel();
-
-					Object o1 = model1.getObject();
-					Object o2 = model2.getObject();
-
-					if (o1.equals(o2)) {
-						return 0;
-					}
-
-					Long cat1;
-					Long cat2;
-					if (o1 instanceof Assignment) {
-						cat1 = ((Assignment) o1).getCategoryId();
-					} else if(o1 instanceof CategoryDefinition) {
-						cat1 = ((CategoryDefinition) o1).getId();
-					} else {
-						return -1;
-					}
-
-					if (o2 instanceof Assignment) {
-						cat2 = ((Assignment) o2).getCategoryId();
-					} else if(o2 instanceof CategoryDefinition) {
-						cat2 = ((CategoryDefinition) o2).getId();
-					} else {
-						return 1;
-					}
-
-					if (cat1 == null) {
-						return 1;
-					} else if (cat2 == null) {
-						return -1;
-					}
-
-					return cat1.compareTo(cat2);
-				}
-			});
-		}
-
 		// TODO make this AjaxFallbackDefaultDataTable
-		final DataTable table = new DataTable("table", cols, studentGradeMatrix, 100);
+		final GbDataTable table = new GbDataTable("table", cols, studentGradeMatrix, 100, Model.ofMap(modelData));
 		table.addBottomToolbar(new NavigationToolbar(table) {
```

Comment:
Hm.. grossly passing in extra data to the custom DataTable.. feeling more like this should be available from the GradebookPage...